cli/command/stack/kubernetes/list.go

```diff
@@ -57,3 +57,96 @@ func getStacks(kubeCli *KubeCli, allNamespaces bool) ([]*formatter.Stack, error)
 	}
 	return formattedStacks, nil
 }
+
+func getStacksWithAllNamespaces(dockerCli command.Cli, opts options.List, kopts Options) ([]*formatter.Stack, error) {
+	stacks, err := getStacks(dockerCli, opts, kopts)
+	if err == nil || !apierrs.IsForbidden(err) {
+		return stacks, err
+	}
+	nms, err2 := getUserVisibleNamespaces(dockerCli)
+	if err2 != nil {
+		logrus.Warnf("Failed to query user visible namespaces: %s", err2)
+		return nil, err
+	}
+	opts.AllNamespaces = false
+	for _, namespace := range nms.Items {
+		kopts.Namespace = namespace.Name
+		ss, err := getStacks(dockerCli, opts, kopts)
+		if err != nil {
+			return nil, err
+		}
+		stacks = append(stacks, ss...)
+	}
+	return stacks, nil
+}
+
+func getUserVisibleNamespaces(dockerCli command.Cli) (*core_v1.NamespaceList, error) {
+	host := dockerCli.Client().DaemonHost()
+	endpoint, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+	endpoint.Scheme = "https"
+	endpoint.Path = "/kubernetesNamespaces"
```

Comment:
UCP does not have a versioned API at present. If we're going to rely on UCP endpoints, we need to be careful about making sure that this stays backwards-compatible.

Is this UX a requirement? Kubernetes does not have any other operations where you can view a list of only the resources that you have access to.

Reply:
You're right, this is really fragile and ad hoc, however this is indeed a UX requirement and Kubernetes does not provide this feature…
If you have any suggestion on how to better manage this I'll be happy to update the code.

Reply:
All right, let's stick with this approach for now, but we'll have to be careful to test this with different combinations of CLI version + UCP version.